core/libp2p/peer/peer_identity.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/peer/peer_identity.hpp"
+
+#include "libp2p/multi/multibase_codec/multibase_codec_impl.hpp"
+
+namespace {
+  constexpr std::string_view kIdSubstr = "/id/";
+
+  const libp2p::multi::MultibaseCodecImpl kCodec{};
+}  // namespace
+
+OUTCOME_CPP_DEFINE_CATEGORY(libp2p::peer, PeerIdentity::FactoryError, e) {
+  using E = libp2p::peer::PeerIdentity::FactoryError;
+  switch (e) {
+    case E::ID_EXPECTED:
+      return "peer id not found in the identity string";
+    case E::SHA256_EXPECTED:
+      return "peer id is not SHA-256 hash";
+    case E::NO_ADDRESSES:
+      return "no addresses in the provided PeerInfo";
+  }
+  return "unknown error";
+}
+
+namespace libp2p::peer {
+  using multi::Multiaddress;
+  using multi::MultibaseCodec;
+  using multi::Multihash;
+
+  PeerIdentity::PeerIdentity(PeerId id, Multiaddress address)
```

Comment:
How do you think, is PeerIdentity good name for this class? 
We already have PeerId and it may confuse users...
Can you come up with a better name?
What if we call it `MultiaddrWithId` ?